test/include/graph_test/graph_test.h

```diff
@@ -66,21 +103,43 @@ class DBTest : public PrivateGraphTest {
                 createConns(connNames);
                 continue;
             }
+            if (statement->connectionsStatusFlag == ConcurrentStatusFlag::BEGIN) {
+                for (auto& concurrentTest : concurrentTests) {
+                    concurrentTest.second.reset();
+                }
+                isConcurrent = true;
+                connectionsPaused = true;
+                continue;
+            }
+            if (statement->connectionsStatusFlag == ConcurrentStatusFlag::END) {
+                for (auto& concurrentTest : concurrentTests) {
+                    concurrentTest.second.execute();
+                }
+                isConcurrent = false;
+                connectionsPaused = false;
+                for (auto& concurrentTest : concurrentTests) {
+                    concurrentTest.second.join();
+                }
+                continue;
+            }
             if (conn) {
                 TestRunner::runTest(statement.get(), *conn, databasePath);
             } else {
                 auto connName = *statement->connName;
-                CheckConn(connName);
-                TestRunner::runTest(statement.get(), *connMap[connName], databasePath);
+                if (isConcurrent) {
+                    concurrentTests.at(connName).addStatement(statement.get());
+                } else {
+                    TestRunner::runTest(statement.get(), *connMap[connName], databasePath);
+                }
             }
         }
     }
 
-    inline void CheckConn(std::string connName) {
-        if (connMap[connName] == nullptr) {
-            connMap[connName] = std::make_unique<main::Connection>(database.get());
-        }
-    }
+protected:
+    bool isConcurrent = false;
+    bool connectionsPaused = false;
```

Comment:
I wonder if you can replace these two bool fields with the enum class ConcurrentStatusFlag.

Reply:
we probably can just keep the bool fields as they make a little more sense to me than the enum